compiler/semtypes.nim

```diff
@@ -1383,7 +1383,7 @@ proc semGeneric(c: PContext, n: PNode, s: PSym, prev: PType): PType =
     return newOrPrevType(tyError, prev, c)
 
   var t = s.typ
-  if t.kind == tyCompositeTypeClass and t.base.kind == tyGenericBody:
+  if t.kind in {tyCompositeTypeClass, tyAlias} and t.base.kind == tyGenericBody:
```

Comment:
Is this related to the bugfix? If so, how?

Reply:
yes it is. without this, `nim c -r tests/metatype/ttypetraits.nim` fails:

ttypetraits.nim(36, 15) Error: no generic parameters allowed for C2b
    doAssert C2b[MyInt, C4[cstring]].name2 == "C[int, C4[cstring]]"


with:
```
type
    C[T0, T1] = object
type C2b=C
```